### PR TITLE
Remove kex_ntru from master (issue #370).

### PR DESCRIPTION
**Pull request by truth-quark**
Redux attempt at removing kex_ntru, starting from master to avoid merge conflicts:
https://github.com/open-quantum-safe/liboqs/issues/370

There are still kex_ntru references in the VisualStudio dir.
@christianpaquin would you mind updating the VisualStudio part on this branch?

Some documentation also needs updates where I've put FIXME placeholders. I'm not sure of what that is to be replaced with.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -143,11 +143,11 @@ There are also a variety of test programs built under the `tests` directory:
 
 To run one or more ciphers only once use `--mem-bench`, which is suitable for memory usage profiling:
 
-	tests/test_kex --mem-bench ntru
+	tests/test_kex --mem-bench FIXME
```
Comment (christianpaquin) [inline README.md]: We shouldn't leave FIXME. In fact, we should remove test_kex altogether.

File: README.md [hunk 2]
```diff
@@ -143,11 +143,11 @@ There are also a variety of test programs built under the `tests` directory:
...
 
-You may also get instant memory usage results of an algorithm (e.g. ntru) by running [valgrind's massif tool](http://valgrind.org/docs/manual/ms-manual.html) by running
+You may also get instant memory usage results of an algorithm (e.g. FIXME) by running [valgrind's massif tool](http://valgrind.org/docs/manual/ms-manual.html) by running
```
Comment (christianpaquin) [inline README.md]: Use something other than FIXME; perhaps the default KEM alg.

File: kex_bench_memory.sh [hunk 3]
```diff
@@ -37,7 +37,7 @@ Usage: $0 [OPTION]... ALGORITHM
   --tmp-dir=DIR       temporary directory [default: $DEFAULT_TMP_DIR]
     ALGORITHM         algorithm to test
 
-Example usage: $0 ntru
+Example usage: $0 FIXME
```
Comment (christianpaquin) [inline kex_bench_memory.sh]: Use something other than FIXME

File: src/kex/kex.c [hunk 4]
```diff
@@ -2,8 +2,6 @@
 
 #include <oqs/kex.h>
```
Comment (christianpaquin) [inline src/kex/kex.c]: We should remove all kex related files at this point (if ntru is the last one)

File: src/kex/kex.h [hunk 5]
```diff
@@ -21,7 +21,6 @@
 
 enum OQS_KEX_alg_name {
```
Comment (christianpaquin) [inline src/kex/kex.h]: We should remove kex.h

File: tests/minimal_kex_oqs.c [hunk 6]
```diff
@@ -17,127 +17,12 @@ void cleanup(uint8_t *alice_msg, size_t alice_msg_len, uint8_t *alice_key,
              uint8_t *bob_key, size_t bob_key_len, void *alice_priv,
              OQS_KEX *kex, OQS_RAND *rnd);
 
-#ifdef ENABLE_KEX_NTRU
```
Comment (christianpaquin) [inline tests/minimal_kex_oqs.c]: We should remove minimal_kex_oqs.c

File: tests/test_kex.c [hunk 7]
```diff
@@ -23,12 +23,6 @@ struct kex_testcase {
 
 /* Add new testcases here */
```
Comment (christianpaquin) [inline tests/test_kex.c]: We should remove test_kex.c

**Comment by truth-quark** [comment]
To keep the NTRU relevant to the ticket, I'm going to remove the KEX mechanism under this ticket:
https://github.com/open-quantum-safe/liboqs/issues/372

File: README.md (hunk 1 above)
Comment (truth-quark) [reply README.md]: I'll remove that in issue: https://github.com/open-quantum-safe/liboqs/issues/372

File: src/kex/kex.c (hunk 4 above)
Comment (truth-quark) [reply src/kex/kex.c]: I'll remove that in issue: https://github.com/open-quantum-safe/liboqs/issues/372

File: src/kex/kex.h (hunk 5 above)
Comment (truth-quark) [reply src/kex/kex.h]: I'll remove it in issue: https://github.com/open-quantum-safe/liboqs/issues/372

File: tests/minimal_kex_oqs.c (hunk 6 above)
Comment (truth-quark) [reply tests/minimal_kex_oqs.c]: I'll remove it in issue: https://github.com/open-quantum-safe/liboqs/issues/372

File: tests/test_kex.c (hunk 7 above)
Comment (truth-quark) [reply tests/test_kex.c]: I'll remove it in issue: https://github.com/open-quantum-safe/liboqs/issues/372

File: README.md (hunk 2 above)
Comment (truth-quark) [reply README.md]: @dstebila have you got any suggestions for a suitable algorithm for these docs?

Comment (dstebila) [reply README.md]: We will have to update the memory benchmark scripts to point to KEM and SIG, so let's remove that for now and at the same time make a corresponding issue to remind us to do it again.

**Comment by dstebila** [comment]
Resolved by #402.

**Pull request closed without merging**